### PR TITLE
StructuredText deserialization and link resolution

### PR DESCRIPTION
**Pull request by loudmouth**

### CODE REVIEW

File: Sources/Contentful/Decodable.swift [hunk 1]
```diff
@@ -93,6 +93,19 @@ internal extension Decodable where Self: AssetDecodable {
...
 }
 
+internal extension Decodable where Self: Node {
+    static func popNodeDecodable(from container: inout UnkeyedDecodingContainer) throws -> Self {
+        do {
+            let contentDecodable = try container.decode(self)
+            return contentDecodable
+
+        } catch let error {
+            print(error)
```
Comment (dlitvakb) [inline Sources/Contentful/Decodable.swift]: Should this print be removed? Seems like a debugging statement

File: Sources/Contentful/Decodable.swift [hunk 2]
```diff
@@ -114,7 +127,7 @@ public extension KeyedDecodingContainer {
      */
     public func resolveLink(forKey key: KeyedDecodingContainer.Key,
                             decoder: Decoder,
-                            callback: @escaping (Any) -> Void) throws {
+                            callback: @escaping (AnyObject) -> Void) throws {
```
Comment (dlitvakb) [inline Sources/Contentful/Decodable.swift]: Is the change from `Any` to `AnyObject` due to the Swift bug you reported yesterday?

File: Sources/Contentful/Entry.swift [hunk 3]
```diff
@@ -79,6 +79,21 @@ public class Entry: LocalizableResource {
                     let resolvedLinks = alreadyResolvedLinks + newlyResolvedLinks
                     resolvedLocalizableFieldMap[localeCode] = resolvedLinks
                 }
+
+                // Resolve links for structured text fields.
+                if let value = fieldValueForLocaleCode as? Document {
+                    let embeddedEntryNodes: [Node] = value.content.map { node in
+                        if let blockNode = node as? EmbeddedEntryBlock {
```
Comment (dlitvakb) [inline Sources/Contentful/Entry.swift]: Pattern Matching 👍

File: Sources/Contentful/Decodable.swift (hunk 2 above)
Comment (loudmouth) [reply Sources/Contentful/Decodable.swift]: yup!

File: Sources/Contentful/Decodable.swift (hunk 1 above)
Comment (loudmouth) [reply Sources/Contentful/Decodable.swift]: ooo good catch, definitely needs to be removed

File: Sources/Contentful/Decodable.swift (hunk 2 above)
Comment (loudmouth) [reply Sources/Contentful/Decodable.swift]: there should be no affect on user code btw, but this is also for v4 which has other breaking changes so shouldn't matter

**Pull request merged**